runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkPipelineTranslator.java

```diff
@@ -27,5 +28,5 @@
   boolean hasTranslation(Class<? extends PTransform<?, ?>> clazz);
 
   <TransformT extends PTransform<?, ?>> TransformEvaluator<TransformT>
-  translate(Class<TransformT> clazz);
+  translate(Class<TransformT> clazz, PCollection.IsBounded isBounded);
```

Comment:
This whole part is a bit messy.. I will take another look.
The main challenge is for streaming pipelines both bounded and unbounded "branches" in the graph. That's also the reason for the streaming translator wrapping the batch as well.